packages/build-tools/src/builders/custom.ts

```diff
@@ -1,13 +1,18 @@
 import path from 'path';
 
-import { BuildPhase, Job } from '@expo/eas-build-job';
-import { BuildConfigParser, BuildStepContext, errors } from '@expo/steps';
+import { BuildPhase, Job, Platform } from '@expo/eas-build-job';
+import { BuildConfigParser, BuildStepContext, errors, BuildPlatform } from '@expo/steps';
 import nullthrows from 'nullthrows';
 
 import { Artifacts, BuildContext } from '../context';
 import { prepareProjectSourcesAsync } from '../common/projectSources';
 import { getEasFunctions } from '../steps/easFunctions';
 
+const platformToCustomBuildPlatform: Record<Platform, BuildPlatform> = {
```

Comment:
I see 2 use cases for feature like that:
- Create task specific for Android/iOS like installing some dependencies like fastlane/cocoapods/ndk
- When setting up sth thtat is installed differently on linux and macos.

The current implementation supports only point 2, but I would argue that we should support only the first point in the config and enforce condition based on the target platform not based on the builder.

I think that if sth is installed differently on Linux and on macOS e.g. apt-get vs brew then the task should detect the platform internally and there is no need to expose option like that to user.

With the current implementation (if we add support for --local) builds if someone runs Android builds on macOS they could not distinguish platform-specific steps using this option.

Reply:
cc @dsokal